tools/validate_spec.py

```diff
@@ -79,81 +78,134 @@ def cli(argv: List[str]) -> Union[int, str]:
         return f"output file already exists: {args.json_output}"
 
     try:
-        validation_main(args)
+        identifier = ID(args.message_identifier)
+    except RecordFluxError as e:
+        return f'invalid identifier "{args.message_identifier}" : {e}'
+
+    try:
+        pdu_message = PyRFLX.from_specs(
+            [str(args.specification)], skip_model_verification=args.no_verification
+        )[str(identifier.parent)][str(identifier.name)]
+    except KeyError:
+        return f'message "{identifier.name}" could not be found in package "{identifier.parent}"'
+
+    except FileNotFoundError as e:
+        return f"specification {e}"
+
+    try:
+        validation_main(
+            pdu_message,
+            args.directory_invalid,
+            args.directory_valid,
+            args.json_output,
+            args.abort_on_error,
+        )
     except ValidationError as e:
         return f"{e}"
 
     return 0
 
 
-def validation_main(args: argparse.Namespace) -> None:
-    # pylint: disable=too-many-locals
-    def get_message_to_validate(message_path: Path, is_valid: bool) -> "OriginalMessage":
-        if not message_path.is_file():
-            raise ValidationError(f"{message_path} is not a regular file")
-        message_bytes = message_path.read_bytes()
-        message = OriginalMessage(message_bytes, is_valid, message_path.name)
-        return message
-
-    root_message_id: str = args.message_identifier
-    path_spec: Path = args.specification
-    path_invalid: Optional[Path] = args.directory_invalid
-    path_valid: Optional[Path] = args.directory_valid
-    full_output_path: Optional[Path] = args.json_output
-    abort_on_error: bool = args.abort_on_error
-    no_verification: bool = args.no_verification
-
-    path_generators: Dict[Path, Iterator[Path]] = {}
-    for path in [path_valid, path_invalid]:
-        if path is not None:
-            path_generators[path] = path.glob("*")
-
+def validation_main(
+    pdu_message: MessageValue,
+    directory_invalid: Optional[Path],
+    directory_valid: Optional[Path],
+    json_output: Optional[Path],
+    abort_on_error: bool,
+) -> None:
```

Comment:
A better name would be nice. This function should be used in `test_specs.py`.

Reply:
Done in 3f95b5a